mlir/include/mlir/IR/AffineMap.h

```diff
@@ -122,9 +122,11 @@ class AffineMap {
   /// `exprs.size()`, as many dims as the largest dim in `exprs` and as many
   /// symbols as the largest symbol in `exprs`.
   static SmallVector<AffineMap, 4>
-  inferFromExprList(ArrayRef<ArrayRef<AffineExpr>> exprsList);
+  inferFromExprList(ArrayRef<ArrayRef<AffineExpr>> exprsList,
+                    MLIRContext *context);
   static SmallVector<AffineMap, 4>
-  inferFromExprList(ArrayRef<SmallVector<AffineExpr, 4>> exprsList);
+  inferFromExprList(ArrayRef<SmallVector<AffineExpr, 4>> exprsList,
+                    MLIRContext *context);
```

Comment:
Nit: I think most APIs take the context as first argument? (thinking of all the attribute / types creation for example)

Reply:
Unfortunately, the codebase isn't following a convention here, and things are being done both ways. Here's a crude search result from `llvm/mlir/`:
```
$ grep  ", context" lib/ -r | wc -l
216
grep  "(context, " lib/ -r | wc -l
247
$ grep  ", context)" lib/IR -r | wc -l
60
$ grep  "(context, " lib/IR -r | wc -l
34
```
From a usage standpoint, I think having it as a trailing argument fits my mental model better, going by "important things come first." So, thinking of the key inputs when opening the parenthesis is natural, followed by the other baggage.